projects/plugins/jetpack/extensions/blocks/nextdoor/edit.js

```diff
@@ -0,0 +1,115 @@
+import { getBlockIconComponent } from '@automattic/jetpack-shared-extension-utils';
+import { getBlockDefaultClassName } from '@wordpress/blocks';
+import { Button, Placeholder, withNotices } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+import './editor.scss';
+import { isEqual } from 'lodash';
+import { getValidatedAttributes } from '../../shared/get-validated-attributes';
+import metadata from './block.json';
+import NextdoorControls from './controls';
+import { parseUrl } from './utils';
+
+const icon = getBlockIconComponent( metadata );
+
+export function NextdoorEdit( {
+	attributes,
+	className,
+	name,
+	noticeOperations,
+	noticeUI,
+	setAttributes,
+} ) {
+	const defaultClassName = getBlockDefaultClassName( name );
+	const validatedAttributes = getValidatedAttributes( metadata.attributes, attributes );
+
+	if ( ! isEqual( validatedAttributes, attributes ) ) {
+		setAttributes( validatedAttributes );
+	}
+
+	const { url } = validatedAttributes;
+	const [ iframeUrl, setIframeUrl ] = useState( url ? parseUrl( url ) : null );
+
+	const [ nextdoorShareUrl, setNextdoorShareUrl ] = useState( iframeUrl );
+
+	const setErrorNotice = () => {
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice(
+			__(
+				"Your Nextdoor post couldn't be embedded. Please double check your URL or code.",
```

Comment:
May be use a generic message?
```suggestion
				"Your Nextdoor post couldn't be embedded. Please double check the URL or code.",
```